src/CommonITILSatisfaction.php

```diff
@@ -0,0 +1,310 @@
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+abstract class CommonITILSatisfaction extends CommonDBTM
+{
+    public $dohistory         = true;
+    public $history_blacklist = ['date_answered'];
+
+    /**
+     * Survey is done internally
+     */
+    public const TYPE_INTERNAL = 1;
+
+    /**
+     * Survey is done externally
+     */
+    public const TYPE_EXTERNAL = 2;
+
+    public static function getTypeName($nb = 0)
+    {
+        return __('Satisfaction');
+    }
+
+    /**
+     * Get the itemtype this satisfaction is for
+     * @return string
+     */
+    public static function getItemtype(): string
+    {
+        // Return itemtype extracted from current class name (Remove 'Satisfaction' suffix)
+        return preg_replace('/Satisfaction$/', '', static::class);
+    }
+
+    /**
+     * for use showFormHeader
+     **/
+    public static function getIndexName()
+    {
+        return static::getItemtype()::getForeignKeyField();
+    }
+
+    public function getLogTypeID()
+    {
+        /** @var CommonITILObject $itemtype */
+        $itemtype = static::getItemtype();
+        return [$itemtype, $this->fields[$itemtype::getForeignKeyField()]];
+    }
+
+    public static function canUpdate()
+    {
+        /** @var CommonITILObject $itemtype */
+        $itemtype = static::getItemtype();
+        return (Session::haveRight($itemtype::$rightname, READ));
+    }
+
+    /**
+     * Is the current user have right to update the current satisfaction
+     *
+     * @return boolean
+     **/
+    public function canUpdateItem()
+    {
+        /** @var CommonITILObject $itemtype */
+        $itemtype = static::getItemtype();
+        $item = new $itemtype();
+        if (!$item->getFromDB($this->fields[$itemtype::getForeignKeyField()])) {
+            return false;
+        }
+
+        // you can't change if your answer > 12h
+        if (
+            !is_null($this->fields['date_answered'])
+            && ((time() - strtotime($this->fields['date_answered'])) > (12 * HOUR_TIMESTAMP))
+        ) {
+            return false;
+        }
+
+        if (
+            $item->isUser(CommonITILActor::REQUESTER, Session::getLoginUserID())
+            || ($item->fields["users_id_recipient"] === Session::getLoginUserID() && Session::haveRight($itemtype::$rightname, Ticket::SURVEY))
```

Comment:
`Ticket::SURVEY` is not the appropriate right for a change.
I guess we have to add a new right for changes.